### ISSUE TITLE
Protected properties (password etc) exposed

### ISSUE THREAD

**Issue by kennethlynne** [strongloop/loopback#1714]
We have configured a relationship where a content belongsTo a user.
Also we have a notification entity that also has a belongsTo relationship to a user.

We use mongoDb.

I see in the database that the owner is saved in its entirety on a content and notification (cached relation). I could not find any documentation on this caching?

# WTF
I also see that when we GET content/:id and GET user/:id/notifications and include owner, it includes all the fields, also the ones specified to be hidden in the user model (password etc).

I guess the metadata about hidden properties is not carried through into the cached relation?

**Comment by AlmaasAre** [comment]
So the issue here is:

When we include the owner of a model, and then we call save() on that model, the owner seems to be stored in MongoDB. Seems like save() does an upsert, and validateUpsert for that model is false.

**Comment by superkhau** [comment]
Can you provide a link to a test project on GitHub? See https://github.com/strongloop/loopback/wiki/Reporting-issues#bug-report

**Comment by kennethlynne** [comment]
The problem was this:

We use the loopback-ds-timestamp-mixin
It disables some validation https://github.com/clarkbw/loopback-ds-timestamp-mixin/blob/master/es6/time-stamp.js#L16

We did some manipulation of the data in some scenarios, after it was loaded with some hydrated relations. We then used the save prototype method on the model to save the changes, effectively saving everything.

The next time it is fetched it included the users password hash also.
